setup.sh

```diff
@@ -140,28 +153,44 @@ chmod a-w ~/.asoundrc
 # Apply changes
 sudo /etc/init.d/alsa-utils restart
 
-# Create the script to run after each reboot and make the soundcard available
-i2s_driver_script=$RPI_SETUP_DIR/resources/load_i2s_driver.sh
-rm -f $i2s_driver_script
+if [[ -n "$I2S_MODE" ]]; then
+    # Create the script to run after each reboot and make the soundcard available
+    i2s_driver_script=$RPI_SETUP_DIR/resources/load_i2s_driver.sh
+    rm -f $i2s_driver_script
 
-# Sometimes with Buster on RPi3 the SYNC bit in the I2S_CS_A_REG register is not set before the drivers are loaded
-# According to section 8.8 of https://cs140e.sergio.bz/docs/BCM2837-ARM-Peripherals.pdf
-# this bit is set after 2 PCM clocks have occurred.
-# To avoid this issue we add a 1-second delay before the drivers are loaded
-echo "sleep 1"  >> $i2s_driver_script
+    # Sometimes with Buster on RPi3 the SYNC bit in the I2S_CS_A_REG register is not set before the drivers are loaded
+    # According to section 8.8 of https://cs140e.sergio.bz/docs/BCM2837-ARM-Peripherals.pdf
+    # this bit is set after 2 PCM clocks have occurred.
+    # To avoid this issue we add a 1-second delay before the drivers are loaded
+    echo "sleep 1"  >> $i2s_driver_script
 
-if [[ -z "$I2S_MODE" ]]; then
-  echo error: I2S mode not known for XMOS device $XMOS_DEVICE.
-  exit 1
-fi
+    I2S_NAME=i2s_$I2S_MODE
+    I2S_MODULE=$RPI_SETUP_DIR/loader/$I2S_NAME/${I2S_NAME}_loader.ko
+    echo "sudo insmod $I2S_MODULE"                            >> $i2s_driver_script
+
+    echo "# Run Alsa at startup so that alsamixer configures" >> $i2s_driver_script	
+    echo "arecord -d 1 > /dev/null 2>&1"                      >> $i2s_driver_script	
+    echo "aplay dummy > /dev/null 2>&1"                       >> $i2s_driver_script
+
+
+    # Create the script to run after each reboot and make the soundcard available
```

Comment:
It looks like lines 176 through 192 duplicate lines 157 through 173.  What's going on here?